src/Core/Utilities/JsonHelpers.cs

```diff
@@ -129,4 +130,63 @@ public override void Write(Utf8JsonWriter writer, DateTime? value, JsonSerialize
             writer.WriteStringValue(CoreHelpers.ToEpocMilliseconds(value.Value).ToString());
         }
     }
+
+    /// <summary>
+    /// Allows reading a string from a number or string, should only be used on string properties
+    /// </summary>
+    public class PermissiveStringConverter : JsonConverter<string>
+    {
+        internal static PermissiveStringConverter Instance = new();
+
+        public override string Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+        {
+            return reader.TokenType switch
+            {
+                JsonTokenType.String => reader.GetString(),
+                JsonTokenType.Number => reader.GetInt64().ToString(),
+                JsonTokenType.Null => null,
+                _ => throw new JsonException($"Unsupported TokenType: {reader.TokenType}"),
+            };
+        }
+
+        public override void Write(Utf8JsonWriter writer, string value, JsonSerializerOptions options)
+        {
+            writer.WriteStringValue(value);
+        }
+    }
+
+    /// <summary>
+    /// Allows reading a string from a number or string, should only be used on string properties
+    /// </summary>
+    public class PermissiveStringEnumerableConverter : JsonConverter<IEnumerable<string>>
+    {
+        public override IEnumerable<string> Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+        {
+            if (reader.TokenType != JsonTokenType.StartArray)
+            {
+                throw new JsonException("Should only be used on a JSON Array");
+            }
+
+            var stringList = new List<string>();
+
+            while (reader.Read() && reader.TokenType != JsonTokenType.EndArray)
+            {
+                stringList.Add(PermissiveStringConverter.Instance.Read(ref reader, typeof(string), options));
```

Comment:
If any of these items fails (throws an exception), is that okay? Should there be handling for that or the entire deserialization should fail? I'm sure this is okay as-is, but just raising the question.

Reply:
I think with the added permissiveness from your other comments I feel comfortable totally throwing. If it ever becomes an issue I would almost rather handle `JsonTokenType.StartArray` and `JsonTokenType.StartObject` in `PermissiveStringConverter` but that goes beyond what Newtonsoft did and not worth it at the moment IMO.